pdata/internal/metrics.go

```diff
@@ -192,19 +192,18 @@ const (
 	MetricDataPointFlagsNone = MetricDataPointFlags(otlpmetrics.DataPointFlags_FLAG_NONE)
 )
 
-// NewMetricDataPointFlags returns a new MetricDataPointFlags combining the flags passed
-// in as parameters.
-func NewMetricDataPointFlags(flags ...MetricDataPointFlag) MetricDataPointFlags {
-	var flag MetricDataPointFlags
-	for _, f := range flags {
-		flag |= MetricDataPointFlags(f)
-	}
-	return flag
+// NewMetricDataPointFlags returns a new MetricDataPointFlags with no flags set.
+func NewMetricDataPointFlags() MetricDataPointFlags {
+	return MetricDataPointFlags(otlpmetrics.DataPointFlags_FLAG_NONE)
+}
+
+// NoRecordedValue returns true if the MetricDataPointFlags contains the NO_RECORDED_VALUE flag.
+func (d MetricDataPointFlags) NoRecordedValue() bool {
+	return d&MetricDataPointFlags(otlpmetrics.DataPointFlags_FLAG_NO_RECORDED_VALUE) != 0
 }
 
-// HasFlag returns true if the MetricDataPointFlags contains the specified flag
-func (d MetricDataPointFlags) HasFlag(flag MetricDataPointFlag) bool {
-	return d&MetricDataPointFlags(flag) != 0
+func (d MetricDataPointFlags) SetNoRecordedValue() MetricDataPointFlags {
```

Comment:
Should accept bool to allow set/unset?

Reply:
Do we use &^ for this?

Reply:
And if that's the case, I'd prefer a Set function and a Remove function. Passing False or True to a Set function makes it seem like it's a bool being set. I'd rather have 2 different functions that describe what they do.

Reply:
> Do we use &^ for this?

That is an implementation detail

> I thought once a flag was bitwise ORed it couldn't be undone.

That may be the case for some flags, but definitely not for all.

> And if that's the case, I'd prefer a Set function and a Remove function. Passing False or True to a Set function makes it seem like it is a bool being set. I'd rather have 2 different functions that describe what they do.

For me the reason to have a "Set" with bool, is because matches the getter. In the end the flag represents a "bool" correct? which can be true of false.

@mx-psi @codeboten @tigrannajaryan any comments on this? What is your preference?

Reply:
I prefer `Set` with a bool, for consistency with other `Set` functions on pdata.

Reply:
Ok, I updated the function to set/remove based on a bool parameter.